src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/QueueProcessingReadinessChecker.java

```diff
@@ -0,0 +1,110 @@
+    ) {
+        this.authenticationChecker = authenticationChecker;
+        this.noAccountLockedCheckValidityDuration = noAccountLockedCheckValidityDuration;
+    }
+
+    /**
+     * Checks if no jurisdiction-specific account is locked in IDAM.
+     * <p>
+     * When an account is locked in IDAM, the processing of the queue must be paused,
+     * in order to let the account get unlocked (through inactivity).
+     * </p>
+     *
+     * @return true if no account is locked in IDAM. Otherwise, throws AccountLockedException in order to
+     *         open the circuit and let Hystrix manage the problem.
+     */
+    @HystrixCommand(
+        commandKey = CHECK_ACCOUNT_NOT_LOCKED_COMMAND_KEY,
+        fallbackMethod = "isNoAccountLockedInIdamFallback"
+    )
+    public boolean isNoAccountLockedInIdam() throws AccountLockedException {
+        try {
+            if (hasNoAccountLockedCheckExpired()) {
+                assertNoAccountIsLockedInIdam();
+            }
+        } catch (AccountLockedException e) {
+            // let hystrix open the circuit
+            throw e;
+        } catch (Exception e) {
+            // just log the exception, but don't let it block queue processing
+            log.error("Failed to check if jurisdictions' accounts are locked in IDAM.", e);
+        }
+
+        return true;
+    }
+
+    private void assertNoAccountIsLockedInIdam() throws AccountLockedException {
+        List<String> jurisdictionsWithLockedAccounts = getJurisdictionsWithLockedAccounts();
+
+        if (jurisdictionsWithLockedAccounts.isEmpty()) {
+            updateNoAccountLockedCheckExpiry();
+        } else {
+            String errorMessage = String.format(
+                "Some jurisdictions' accounts are locked in IDAM. "
+                    + "This will pause queue processing. Jurisdictions with locked accounts: [%s]",
+                String.join(",", jurisdictionsWithLockedAccounts)
+            );
+
+            log.error(errorMessage);
+
+            // open the circuit
+            throw new AccountLockedException(errorMessage);
+        }
+    }
+
+    public boolean isNoAccountLockedInIdamFallback() {
+        log.warn("Executing fallback method for {} command", CHECK_ACCOUNT_NOT_LOCKED_COMMAND_KEY);
+        return false;
+    }
+
+    private List<String> getJurisdictionsWithLockedAccounts() {
+        return authenticationChecker
+            .checkSignInForAllJurisdictions()
+            .stream()
+            .filter(status -> Objects.equals(status.errorResponseStatus, LOCKED.value()))
+            .map(status -> status.jurisdiction)
+            .collect(toList());
+    }
+
+    private boolean hasNoAccountLockedCheckExpired() {
+        return !LocalDateTime.now().isBefore(noAccountLockedCheckExpiry);
+    }
+
+    private void updateNoAccountLockedCheckExpiry() {
+        this.noAccountLockedCheckExpiry =
+            LocalDateTime.now().plus(noAccountLockedCheckValidityDuration);
```

Comment:
consider using a comment if this oneliner is not clear, instead of moving it to a method that's called once.

Reply:
I think extracting this to a separate method is the cleanest. The calling method (`assertNoAccountIsLockedInIdam`) is focused on a different concern. Placing this logic directly there would make it just slightly less readable.

Reply:
I've just moved the call outside the assertion, as that wasn't the best place for it.